src/pymor/operators/basic.py

```diff
@@ -54,23 +55,20 @@ def assemble(self, mu=None):
             return self
 
     def __sub__(self, other):
-        if isinstance(other, Number):
-            assert other == 0.
-            return self
+        assert isinstance(other, OperatorInterface)
         from pymor.operators.constructions import LincombOperator
         return LincombOperator([self, other], [1, -1])
 
     def __add__(self, other):
-        if isinstance(other, Number):
-            assert other == 0.
-            return self
+        assert isinstance(other, OperatorInterface)
         from pymor.operators.constructions import LincombOperator
-        return LincombOperator([self, other], [1, 1])
-
-    __radd__ = __add__
+        if isinstance(other, LincombOperator):
+            return other + self
+        else:
+            return LincombOperator([self, other], [1., 1.])
```

Comment:
Nitpick: `[1, 1]` became `[1., 1.]` here, while there is `[1, -1]` in `__sub__`.